config/v1/types_node.go

```diff
@@ -46,6 +46,17 @@ type NodeSpec struct {
 	// the status and corresponding reaction of the cluster
 	// +optional
 	WorkerLatencyProfile WorkerLatencyProfileType `json:"workerLatencyProfile,omitempty"`
+
+	// MinimumKubeletVersion is the lowest version of a kubelet that can meaningfully join the cluster.
+	// Specifically, the apiserver will deny most authorization requests of kubelets that are older
+	// than the specified version, only allowing the kubelet to get and update its node object, and perform
+	// subjectaccessreviews.
+	// Its max length is 8, so maximum version allowed is either "9.999.99" or "99.99.99".
+	// +kubebuilder:validation:XValidation:rule="self.matches('^[0-9]*.[0-9]*.[0-9]*$')",message="minmumKubeletVersion must be in a semver compatible format of x.y.z"
+	// +kubebuilder:validation:MaxLength:=8
+	// +openshift:enable:FeatureGate=MinimumKubeletVersion
+	// +optional
+	MinimumKubeletVersion string `json:"minimumKubeletVersion"`
```

Comment:
Is this Kubernetes versioning, or OCP versioning?

What would happen if the version string in the payload had a suffix, `1.2.3-ec.0`?

Reply:
kubernetes, and it would fail (despite technically being valid semver). I thought about allowing it but it doesn't feel useful to me personally

Reply:
Should the godoc specify that this is the Kubernetes version, specifically? So it's not confused later with OCP version?

Have you checked with the workloads folks that the reported K8s version is always going to be without a suffix? I'm wondering if that's true during a rebase and whether restricting versions with suffices would break the rebase somehow?

Reply:
so the version of the kubelet can have a suffix, as [it'll be stripped](https://github.com/openshift/kubernetes/pull/2104/files#diff-b7bdcb41a84b0a21cc83959481df9b457492b2e10b6d1350dff1278c8489bf44R91-R92) when comparing. The only thing that can't is the API. If a cluster needs to set the minimum version to an RC, they can set it to 1.x.0, and that'll include ecs

Reply:
Is that worth capturing in the godoc for this field?

Reply:
good idea, I'll add a note

Reply:
updated!